packages/web/src/pages/audio-page/components/modals/ConnectedWalletsModal.tsx

```diff
@@ -0,0 +1,232 @@
+import { useCallback, useContext, useState } from 'react'
+
+import {
+  useConnectedWallets,
+  useRemoveConnectedWallet
+} from '@audius/common/api'
+import { Chain } from '@audius/common/models'
+import { useConnectedWalletsModal } from '@audius/common/store'
+import {
+  Button,
+  Flex,
+  IconWallet,
+  LoadingSpinner,
+  Modal,
+  ModalContentPages,
+  ModalFooter,
+  ModalHeader,
+  ModalTitle,
+  Text
+} from '@audius/harmony'
+
+import { ToastContext } from 'components/toast/ToastContext'
+import { reportToSentry } from 'store/errors/reportToSentry'
+import { NEW_WALLET_CONNECTED_TOAST_TIMEOUT_MILLIS } from 'utils/constants'
+
+import {
+  AlreadyAssociatedError,
+  useConnectAndAssociateWallets
+} from '../../../../hooks/useConnectAndAssociateWallets'
+import WalletsTable, { WalletTableRow } from '../WalletsTable'
+
+export const WALLET_COUNT_LIMIT = 5
+
+const messages = {
+  title: 'Connected Wallets',
+  description:
+    'Connect wallets to your account to display external $AUDIO balances and showcase NFT collectibles on your profile.',
+  connect: 'Connect Wallet',
+  limit: `Reached Limit of ${WALLET_COUNT_LIMIT} Connected Wallets.`,
+  noConnected: 'You haven’t connected any wallets yet.',
+  back: 'Back',
+  newWalletConnected: 'New Wallet Successfully Connected!',
+  walletRemoved: 'Wallet Successfully Removed!',
+  warning: 'Are you sure you want to remove this wallet from your account?',
+  remove: 'Remove Wallet',
+  ignore: 'Nevermind',
+  error: 'Something went wrong. Please try again.',
+  walletAlreadyAdded: 'No new wallets selected to connect.'
+}
+
+enum Pages {
+  TABLE = 0,
+  CONFIRM_REMOVE_WALLET = 1
+}
+
+export const ConnectedWalletsModal = () => {
+  const { isOpen, onClose, onClosed } = useConnectedWalletsModal()
+  const { toast } = useContext(ToastContext)
+
+  const [currentPage, setCurrentPage] = useState(Pages.TABLE)
+  const [walletToRemove, setWalletToRemove] = useState<{
+    address: string
+    chain: Chain
+  }>()
+
+  const {
+    data: connectedWallets,
+    isPending,
+    isError,
+    error
+  } = useConnectedWallets()
+
+  const {
+    mutateAsync: removeConnectedWalletAsync,
+    isPending: isRemovePending
+  } = useRemoveConnectedWallet()
+
+  const handleRemoveClicked = useCallback(
+    (wallet: { address: string; chain: Chain }) => {
+      setWalletToRemove(wallet)
+      setCurrentPage(Pages.CONFIRM_REMOVE_WALLET)
+    },
+    []
+  )
+
+  const handleRemoveConfirmed = useCallback(async () => {
+    try {
+      await removeConnectedWalletAsync({ wallet: walletToRemove! })
+      setCurrentPage(Pages.TABLE)
+      toast(messages.walletRemoved, NEW_WALLET_CONNECTED_TOAST_TIMEOUT_MILLIS)
+    } catch (e) {
+      if (e instanceof Error) {
```

Comment:
why add this check?

Reply:
LOL good q - originally was putting into reportToSentry I bet, but yeah not needed